packages/node_modules/@webex/plugin-logger/src/logger.js

```diff
@@ -195,34 +209,94 @@ const Logger = WebexPlugin.extend({
     }
 
     return 'error';
-  }
-});
-
-levels.forEach((level) => {
-  let impls = fallbacks[level];
-  let impl = level;
+  },
 
-  if (impls) {
-    impls = impls.slice();
-    // eslint-disable-next-line no-console
-    while (!console[impl]) {
-      impl = impls.pop();
+  /**
+   * Format logs (for upload)
+   *
+   * If separate client, SDK buffers is configured, merge the buffers, if configured
+   *
+   * @instance
+   * @memberof Logger
+   * @public
+   * @memberof Logger
+   * @returns {string} formatted buffer
+   */
+  formatLogs() {
+    // TODO: revisit the type of date field - keep as Date until format stage? that could make merge more efficient
+    function getDate(log) {
+      return log[0];
+    }
+    const buffer = [];
+    let clientIndex = 0;
+    let sdkIndex = 0;
+
+    // TODO: if (new merge method) {
+    // merge the two buffers
+    // while we have entries in either buffer
+    while (clientIndex < this.clientBuffer.length || sdkIndex < this.sdkBuffer.length) {
+      // if we have remaining entries in the SDK buffer
+      if (sdkIndex < this.sdkBuffer.length &&
+          // and we haven't exhausted all the client buffer entries, or SDK date is before client date
+          (clientIndex > this.clientBuffer.length ||
+            (new Date(getDate(this.sdkBuffer[sdkIndex])) <= new Date(getDate(this.clientBuffer[clientIndex]))))) {
+        // then add to the SDK buffer
+        buffer.push(this.sdkBuffer[sdkIndex]);
+        sdkIndex += 1;
+      }
+      // otherwise if we haven't exhausted all the client buffer entries, add client entry, whether it was because
+      // it was the only remaining entries or date was later (the above if)
+      else if (clientIndex < this.clientBuffer.length) {
+        buffer.push(this.clientBuffer[clientIndex]);
+        clientIndex += 1;
+      }
     }
+    // } end if merge...
+    // else { ie old way
+    //   buffer = this.buffer;
+    // }
+
+    return buffer.join('\n');
   }
+});
 
+/**
+ * Creates a logger method
+ *
+ *
+ * @param {string} level level to create (info, error, warn, etc.)
+ * @param {string} impl the level to use when writing to console
+ * @param {string} type type of log, SDK or client
+ * @param {string} neverPrint function never prints to console
+ * @param {string} alwaysBuffer function always logs to log buffer
+ * @instance
+ * @memberof Logger
+ * @private
+ * @memberof Logger
+ * @returns {function} logger method with specified params
+ */
+function makeLoggerMethod(level, impl, type, neverPrint = false, alwaysBuffer = false) {
```

Comment:
I noticed `neverPrint` and `alwaysBuffer` are either both false or both true. Would there ever be a case were this isn't true?

Reply:
I offered neverPrint separately because I figure it should only be used in extreme cases.  In the Teams web client case, we are overriding console methods and sending them to the buffered logs.  In that case, a console print would be a severe problem, since it would cause infinite recursion.  So we want to suppress any possibility of a console log.  Other cases might not want to suppress logging so severely, although, true, I didn't implement such methods.

Reply:
@mwegman Are you OK with resolving this conversation?